README.md

```diff
@@ -1,12 +1,267 @@
+# Syndesis
+
+Syndesis is a single page application built with React.
+
+## Table of Contents
+
+* [Architecture](#architecture)
+  * [syndesis](#syndesis-1)
+  * [packages](#packages)
+      * [api](#packagesapi)
+      * [models](#packagesmodels)
+      * [ui](#packagesui)
+      * [utils](#packagesutils)
+      * [syndesis-context](#packagessyndesis-context)
+  * [typings](#typings)
+* [Installation](#installation)
+* [Building](#building)
+* [Scripts](#scripts)
+* [Roadmap](#roadmap)
+
+## Architecture 
+
+We use [Lerna](https://github.com/lerna/lerna) to streamline the development process; the most common operations, 
+like building the project or running the development mode can be done directly from the project root.  
+
+Code is split in many packages, organized as a monorepo using [Yarn's workspaces](https://yarnpkg.com/lang/en/docs/workspaces/). 
+The workspace is configured like this:
+
 ```
+"syndesis",
+"packages/*"
+"typings/*"
+```
+
+### syndesis
+
+`syndesis` is the main application. It handles the authentication against Syndesis's OAuth Server, and provides the main
+ app layout where "sub-apps" can be injected. 
+
+![](doc/assets/syndesis-chrome.png)
+
+It also provides an [API for sub-apps](#packagessyndesis-context) - in the form of a [React's Context](https://reactjs.org/docs/context.html) - for 
+interact with it, eg. closing the navigation bar, or redirecting to another sub-app.
+
+It's built with [create-react-app](https://github.com/facebook/create-react-app).
+
+#### Development server
+
+```bash
+# From the repository root
+$ yarn watch:app
+```
+
+Or
+
+```bash
+# From the syndesis folder
+$ yarn start
+```
+
+
+#### First time setup
+
+The app requires some extra configuration on your Syndesis installation in order to work.
+
+We need to create an OAuth Client to allow the UI to login against Syndesis OAuth Server, and then expose the 
+server with a direct route.
+
+First, login as an "admin" user of your cluster:    
+```bash
+$ oc login -u admin
+```
+
+Then, create a new OAuth Client:
+```bash
 oc create -f <(echo '
 kind: OAuthClient
 apiVersion: oauth.openshift.io/v1
 metadata:
  name: camel-k-ui
 secret: "..."
 redirectURIs:
- - "http://localhost:3000"
+ - "http://localhost:5000/"
 grantMethod: prompt
 ')
-```
\ No newline at end of file
+```
+
+Finally, expose the server with a direct route.   
+**Replace the `CLUSTER_ADDRESS` placeholder with the right address of your cluster.**
+For a local development environment running on Minishift, it will look like this: `192.168.64.1.nip.io` 
```

Comment:
I made a mistake and left off the `.nip.io` extension the first time around. When I re-ran the same command, just with the proper url this time, I get the following error:

`Error from server (AlreadyExists): error when creating "/dev/fd/63": routes.route.openshift.io "syndesis-server" already exists`.

I imagine this may happen to others as well, can we inform how to see all available routes and how to delete the old route we've exposed?

Reply:
I think shortly after merging this we'll automate this stuff via a script to be honest, as having to run these commands by hand is error prone.